zcash_primitives/src/group_hash.rs

```diff
@@ -43,3 +43,30 @@ pub fn group_hash<E: JubjubEngine>(
         Err(_) => None,
     }
 }
+
+pub fn find_group_hash<E: JubjubEngine>(
+    m: &[u8],
+    personalization: &[u8; 8],
+    params: &E::Params
+) -> edwards::Point<E, PrimeOrder>
+{
+    let mut tag = m.to_vec();
+    let i = tag.len();
+    tag.push(0u8);
+
+    loop {
+        let gh = group_hash(
+            &tag,
+            personalization,
+            params
+        );
+
+        // We don't want to overflow and start reusing generators
+        assert!(tag[i] != u8::max_value());
+        tag[i] += 1;
+
+        if let Some(gh) = gh {
```

Comment:
What fraction of attempts should pass this check (in expectation)? We should write a comment about this.

Reply:
I had to think about this one a bit. The jubjub curve is not itself a prime order group, but rather composite. I think the probability that a random point lands in the small order (cofactor) subgroup is really small, so the expected number of loops should be very close to 1

Reply:
Alright, let's add that in a comment.

Reply:
Ah, I forgot one thing. In addition to the (small) probability of the hash landing on a small order, only approximately 50% of field elements occur as y coordinates on an Elliptic curve. So the blake2 hash might not hit a curve point at all, so we need to loop until the hash hits a point to try. Since the hash images can be roughly treated as independent, the last byte of the preimage can be used to find a point. The probability of all 256 last possible bytes not hitting any curve points is on the order of ~2^-256 so unlikely to occur.